#Parameters named as in
#Optimization-Based Autonomous Racingof 1:43 Scale RC Cars
#Alexander Liniger, Alexander Domahidi and Manfred Morari
# https://onlinelibrary.wiley.com/doi/abs/10.1002/oca.2123
m: 2.99 #[kg]
l_f: 0.15 #[m]
l_r: 0.107 #[m]
Iz:  0.1 #[kg*m^3]

#pajecka and motor coefficients
B_f: 8.8931
B_r: 45.4263
C_f: 1.3781
C_r: 0.3558
D_f: 12.8523
D_r: 12.8523

C_m1: 30.5921
C_m2: 0.7308
C_roll: 0.0822
C_d: 0.6420

# box constraints
# velocity
v_max: 3.5 # max long vel [m/s]
v_min: 0 #0.05 # min long vel [m/s]

# steering
delta_min: -0.35  # minimum steering angle [rad]
delta_max: 0.35  # maximum steering angle [rad]

deltadot_min: -5.5  # minimum steering angle cahgne[rad/s]
deltadot_max: 5.5 # maximum steering angle cahgne[rad/s]

d_min: -1.0
d_max: 1.0

a_min: -5.0
a_max: 5.0

# # # Weight
# # # good traking
# Q_c: 100 # speed
# Q_l: 1000 # a lrage value
# Q_theta: 1

# # too small is bad
# R_d: 0.001
# R_delta: 0.01 


Q_c: 100 # cost for contouring error
Q_l:  100 # lag error cost
Q_theta: 10 # advancing parameter reward
R_d: 0.05
R_delta: 0.05
